clang/lib/ExtractAPI/ExtractAPIConsumer.cpp

```diff
@@ -793,7 +793,7 @@ void ExtractAPIAction::EndSourceFileAction() {
   // FIXME: Make the kind of APISerializer configurable.
   SymbolGraphSerializer SGSerializer(*API, ProductName);
   SGSerializer.serialize(*OS);
-  OS->flush();
+  OS = nullptr;
```

Comment:
I feel like this deletion is not intuitive when you try to write a FrontendAction, while `flush()` feels more natural. Not a blocker but it would be more natural if we can layer the CompilerInstance and FrontendAction in a way that `clearOutputFile` is only called after Action is deleted.

Reply:
I do think deleting the stream, while not immediately intuitive, is better style overall:
- `flush()` means push out anything pending; not necessarily the last write though.
- `reset()` (or `= nullptr`) means there's nothing more to write.

But I fully agree with you that layering to have the destruction happen naturally would be ideal (even though I'm not sure that's practical for clang, where `clearOutputFiles()` can be called at other times).